fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -365,6 +368,47 @@ public void updateUniquenessViolations(@Nonnull Tuple valueKey, @Nonnull Tuple p
         return keyValues.map(kv -> unpackKeyValue(uniquenessViolationsSubspace, kv));
     }
 
+    /**
+     * Validate the integrity of the index (such as identifying index entries that do not point to records or
+     * identifying records that do not point to valid index entries. The default implementation provided by the
+     * <code>StandardIndexMaintainer</code> class is a no-op (performs no validation) and should be overridden by
+     * implementing classes.
+     * @param continuation any continuation from a previous validation invocation
+     * @param scanProperties skip, limit and other properties of the validation (Use default ones if <code>null</code>)
+     * @return an empty cursor.
+     */
+    @Nonnull
+    @Override
+    public RecordCursor<InvalidIndexEntry> validateEntries(@Nullable byte[] continuation,
+                                                           @Nullable ScanProperties scanProperties) {
+        return RecordCursor.empty();
+    }
+
+    /**
+     * Validate entries in the index. It scans the index and checks if the record associated with each index entry exists.
+     * @param continuation any continuation from a previous validation invocation
+     * @param scanProperties skip, limit and other properties of the validation (Use default ones if <code>null</code>)
+     * @return a cursor over index entries that have no associated records.
+     */
+    RecordCursor<InvalidIndexEntry> validateOrphanEntries(@Nullable byte[] continuation,
```

Comment:
Missing nullability annotation on return type

Reply:
Also, I assume package-private with Javadoc is intentional even though it needs to be at least `protected` to be rendered? I'm fine with leaving it package private for now and maybe making it protected in the future unless there is a thought that external `IndexMaintainer`s might want this.

Reply:
Added `@Nonnull`.

External `StandardIndexMaintainer` should be able to use it. So Added `protected`.